TodoApp/src/TodoApp.Api/Controllers/TodosController.cs

```diff
@@ -45,19 +42,24 @@ public TodosController(ITodoRepository todoRepository, IUriHelper uriHelper)
         {
             var newTodo = await _repository.CreateAsync(todo);
 
-            var location = _uriHelper.BuildUriForPostTodo(newTodo.Id);
+            var location = _uriHelper.BuildRouteUri(newTodo.Id);
 
-            return Created(location, newTodo);          
+            return Created(location, newTodo);
         }
 
         public async Task<IHttpActionResult> DeleteTodoAsync(Guid id)
         {
+            if (await _repository.RemoveAsync(id))
+                return await Task.FromResult(StatusCode(HttpStatusCode.NoContent));
+
             return await Task.FromResult(StatusCode(HttpStatusCode.NoContent));
         }
 
         public async Task<IHttpActionResult> PutTodoAsync(Guid id, [FromBody] Todo updated)
         {
-            return await Task.FromResult(Ok(updated));
+            var newTodo = await _repository.UpdateAsync(updated);
```

Comment:
The result is actually just a bool according to your interface. It fails a test.